examples/allennlp/subsample_dataset_reader.py

```diff
@@ -0,0 +1,26 @@
+import random
+
+import allennlp
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from sklearn.model_selection import train_test_split
+
+
+@DatasetReader.register("subsample")
+class SubsampledDatasetReader(allennlp.data.dataset_readers.TextClassificationJsonReader):
```

Comment:
Thank you for the review.
Initially, I create `SubSampledDataset`, which describes a reader for a subsampled dataset.
But I renamed it to `SubsampleDataset` since I could describe this class as a sub-sample dataset.
(And I thought it would be better to use only present tense words in the filenames if it was possible.)

I'm not confident for this convention, so I'd like to get your feedback.
Do you think `SubsampledDatasetReader` is better?

Reply:
I think either one is OK because I found both use cases.

- `SubsampledDatasetReader` can be seen in the following issue:
https://github.com/allenai/allennlp/issues/3099#issuecomment-515688316
(BTW, [`itertools.islice`](https://docs.python.org/ja/3/library/itertools.html#itertools.islice) was employed in the code example in the issue. It can be useful if the dataset is too large to fit in memory.)
- `subsample` is also used in the variable name [`batch_subsample_map`](https://github.com/allenai/allennlp/issues/1408#issuecomment-401143900).

I prefer `SubsampleDatasetReader` for its simplicity, but it is not a strong opinion.

Reply:
OK, so let me leave the dataset reader name as is.

> `islice` ... It can be useful if the dataset is too large to fit in memory.
Thank you for pointing out, you're right!
I gave it try to use `islice` in 4437f48 and found the dataset is quite balanced.

```
train: Counter({'0': 1003, '1': 997})
validation: Counter({'1': 489, '0': 511})
```

I also ran HPO on GPU to confirm training could succeed.

```
python allennlp_simple.py
....
[I 2020-07-22 13:38:58,896] Trial 21 finished with value: 0.704 and parameters: {'dropout': 0.18415084463564746, 'embedding_dim': 33, 'output_dim': 69, 'max_filter_size': 4, 'num_filters': 55, 'lr': 0.09093981266227305}. Best is trial 12 with value: 0.725.
Number of finished trials:  22
Best trial:
  Value:  0.725
  Params:
    dropout: 0.31246648830863727
    embedding_dim: 22
    output_dim: 98
    max_filter_size: 4
    num_filters: 38
    lr: 0.09838055114180111
```

So I determined to remove `train_test_split` and simply use `islice`.